From 42bd893aeeac109d276f5cd85c24fa7e9944650d Mon Sep 17 00:00:00 2001
Message-Id: <42bd893aeeac109d276f5cd85c24fa7e9944650d.1594934306.git.Jim.Somerville@windriver.com>
From: eric zhang <eric.zhang@windriver.com>
Date: Wed, 14 Mar 2018 14:56:21 -0400
Subject: [PATCH 1/1] Add build dependencies and package versioning for TiS
 format

Force disable python3 to avoid random build failures due to
using python3 path.

Signed-off-by: eric zhang <eric.zhang@windriver.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Shuicheng Lin <shuicheng.lin@intel.com>
---
 SPECS/rdma-core.spec | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/SPECS/rdma-core.spec b/SPECS/rdma-core.spec
index d040b59..b42393f 100644
--- a/SPECS/rdma-core.spec
+++ b/SPECS/rdma-core.spec
@@ -7,6 +7,12 @@
 %endif
 %endif
 
+%if "%{_tis_build_type}" == "rt"
+%define bt_ext -rt
+%else
+%undefine bt_ext
+%endif
+
 %{!?cmake: %global cmake cmake}
 %{!?make_jobs: %global make_jobs make VERBOSE=1 %{?_smp_mflags}}
 %{!?cmake_install: %global cmake_install DESTDIR=%{buildroot} make install}
@@ -35,7 +41,7 @@
 
 Name: rdma-core
 Version: 50mlnx1
-Release: 1%{?dist}.50218
+Release: 1.50218%{?_tis_dist}.%{tis_patch_ver}
 Summary: RDMA core userspace libraries and daemons
 Group: System Environment/Libraries
 
@@ -91,6 +97,9 @@ BuildRequires: python3
 BuildRequires: python
 %endif
 %endif
+BuildRequires: cmake
+BuildRequires: libnl3-devel
+BuildRequires: mlnx-ofa_kernel%{?bt_ext}-devel
 
 %if %{with build_docs}
 %if 0%{?rhel} >= 8 || 0%{?fedora} >= 30 || %{with_pyverbs}
@@ -397,10 +406,6 @@ easy, object-oriented access to IB verbs.
          -DENABLE_STATIC=1 \
 %endif
          %{EXTRA_CMAKE_FLAGS} \
-%if %{defined __python3}
-         -DPYTHON_EXECUTABLE:PATH=%{__python3} \
-         -DCMAKE_INSTALL_PYTHON_ARCH_LIB:PATH=%{python3_sitearch} \
-%endif
 %if %{with srp_compat}
          -DENABLE_SRP_COMPAT=1 \
 %endif
-- 
1.8.3.1

